benchmark-sets/structured/abseil-cpp.yaml

```diff
@@ -0,0 +1,69 @@
+"functions":
+- "name": "boolabsl::str_format_internal::FormatArgImpl::Dispatch<longlong>(absl::str_format_internal::FormatArgImpl::Data,absl::str_format_internal::FormatConversionSpecImpl,void*)"
```

Comment:
I think this depends a bit on what we're going for.

Is the function what one reads or what the compiler uses as the unique identifier for the function? If it's the latter, then perhaps using the mangled version is the right choice, which will include e.g. types etc.

Alternatively, is the function name:

- `absl::str_format_internal::FormatArgImpl::Dispatch<longlong>`
- `Dispatch`
- `Dispatch<longlong>`
...

?

I think likely the mangled is the right choice for uniquely identifying CPP functions, and what is used atm is the demangled version stripped from whitespace.

From a program analysis perspective, I think we will end up using the mangled version as the function name (this is what you'll see in the debugger if gdb'ing a C++ program), as this is the unique name identifier. However, it likely means we need some heuristics when queries are done based on the demangled name.

Reply:
For reference,

The two `rfind` functions defined here: https://github.com/abseil/abseil-cpp/blob/ddcf8be90575d494e40fcd4e0c408f0237efe0da/absl/strings/string_view.h#L485-L492 provide the following debug information:

```
{
    "name": "rfind",
    "source": {
        "source_file": "/proc/self/cwd/external/com_google_absl/absl/strings/string_view.h",
        "source_line": "485"
    },
    "args": [
        "const string_view *",
        "char",
        "size_type"
    ],
    "return_type": "size_type"
},
{
    "name": "rfind",
    "source": {
        "source_file": "/proc/self/cwd/external/com_google_absl/absl/strings/string_view.h",
        "source_line": "489"
    },
    "args": [
        "const string_view *",
        "Nonnull<const char *>",
        "size_type",
        "size_type"
    ],
    "return_type": "size_type"
},
```

Perhaps what we need is two keys per function: `fuction_identifier`, `function_name`?

Reply:
Based on our Slack channel discussion, I suppose we will use the function signature from the developer's view (i.e., no `bool` before function name) and use function signature as the unique identifier?

Reply:
Added [a temp fix](https://github.com/google/oss-fuzz-gen/pull/64/commits/5947c70eff1bd3df8934e8fd2e850523fb1adaf8#diff-2988c9b2abbf23c03fa8ed236ed0f62fb249e13ef31c1509b7fef0f6d78dd1beL201-R221) for now, but ideally, this can be handled by FI : )